controllers/storagecluster/consumer.go

```diff
@@ -0,0 +1,168 @@
+		!instance.Status.ExternalStorage.GrantedCapacity.Equal(*instance.Spec.ExternalStorage.RequestedCapacity) {
+		return updateConsumerCapacity(r, instance)
+	}
+
+	return nil
+}
+
+// ensureDeleted performs offboarding
+func (o *ocsConsumer) ensureDeleted(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+
+	if !isExternalOCSProvider(instance) {
+		return nil
+	}
+
+	return offBoardConsumer(r, instance)
+}
+
+// onBoardConsumer makes an API call to the provider cluster for onboarding
+func onBoardConsumer(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+
+	// TODO: think about naming convention
+	response, err := r.providerClient.OnboardConsumer("my-ocs", getExternalStorageRequestedCapacity(instance))
+	err = r.handleOnboardConsumerErrors(instance, err)
+	if err != nil {
+		return err
+	}
+
+	if response.StorageConsumerUUID == "" || response.GrantedCapacity == "" {
+		err = fmt.Errorf("External-OCS:OnboardConsumer:response is empty")
+		r.Log.Error(err, "empty response")
+		return err
+	}
+
+	instance.Status.ExternalStorage.ConsumerID = response.StorageConsumerUUID
+	instance.Status.ExternalStorage.GrantedCapacity = resource.MustParse(response.GrantedCapacity)
+
+	return nil
+}
+
+// offBoardConsumer makes an API call to the provider cluster for offboarding
+func offBoardConsumer(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+
+	_, err := r.providerClient.OffboardConsumer(instance.Status.ExternalStorage.ConsumerID)
+	return r.handleOffboardConsumerErrors(instance, err)
+}
+
+// upscaleOrDownscaleConsumer makes an API call to the provider cluster to update the capacity
+func updateConsumerCapacity(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+
+	response, err := r.providerClient.UpdateCapacity(
+		instance.Status.ExternalStorage.ConsumerID,
+		getExternalStorageRequestedCapacity(instance))
+	err = r.handleUpdateConsumerCapacityErrors(instance, err)
+	if err != nil {
+		return err
+	}
+
+	if response.GrantedCapacity == "" {
+		err = fmt.Errorf("External-OCS:UpdateCapacity:response is empty")
+		r.Log.Error(err, "empty response")
+		return err
+	}
+
+	instance.Status.ExternalStorage.GrantedCapacity = resource.MustParse(response.GrantedCapacity)
+
+	return nil
+}
+
+// getExternalConfigFromProvider makes an API call to the provider cluster for json blob
+// and save it in a global var (externalOCSResources) to use the same json again and again
+// It also skips the API call if json is already available
+func getExternalConfigFromProvider(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) ([]ExternalResource, error) {
+
+	if externalOCSResources[instance.UID] != nil {
+		return externalOCSResources[instance.UID], nil
+	}
+
+	config, err := r.providerClient.GetStorageConfig(instance.Status.ExternalStorage.ConsumerID)
+	err = r.handleGetStorageConfigErrors(instance, err)
+	if err != nil {
+		return nil, err
+	}
+
+	var externalResources []ExternalResource
```

Comment:
How about GetStorageConfig returned ExternalResource and not a json?

Reply:
No it does not returns `ExternalResource` it returns `Data []byte `protobuf:"bytes,1,opt,name=data,proto3" json:"data,omitempty"`` in the response